autoload/ale/code_action.vim

```diff
@@ -106,10 +109,17 @@ function! ale#code_action#ApplyChanges(filename, changes) abort
         call remove(l:lines, -1)
     endif
 
-    call ale#util#Writefile(l:buffer, l:lines, a:filename)
+    if a:should_save
+        call ale#util#Writefile(l:buffer, l:lines, a:filename)
+    else
+        call ale#fix#SetBufferContents(l:buffer, l:lines)
```

Comment:
Move the function to the `util.vim` file so the `code_action.vim` code doesn't load the `fix.vim` code, if `:ALEFix` is never used.